### PR TITLE
D8CORE-000 Add filter to render and clean an array

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Added a filter that does the rendering, strip tags, and all the stuff to keep special characters and things.

# Need Review By (Date)
- 11/4

# Urgency
- medium

# Steps to Test
1. checkout this branch
1. clear caches
1. view your favorite pages and verify they look the same.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: src/Plugin/TwigPlugin/JumpstartUITwig.php [hunk 1]
```diff
@@ -2,36 +2,89 @@
...
+   */
+  public function renderClean($elements, $tags = '<drupal-render-placeholder>') {
+    if ($elements instanceof MarkupInterface) {
+      $elements = ['#markup' => (string) $elements];
+    }
+
+    $rendered = $this->renderer->render($elements);
+    if (strpos($tags, '<drupal-render-placeholder>') === FALSE) {
+      $tags .= '<drupal-render-placeholder>';
+    }
```
Comment (sherakama) [inline src/Plugin/TwigPlugin/JumpstartUITwig.php:84]: 🤔 Is there a reason why we would want to strip everything including the placeholder?

Comment (sherakama) [reply src/Plugin/TwigPlugin/JumpstartUITwig.php:84]: The question I am getting at is do we want to add the placeholder at all times or should we be more explicit. I usually lean towards explicit.

File: tests/src/Unit/Plugin/TwigPlugin/JumpstartUITwigTest.php [hunk 2]
```diff
@@ -58,6 +63,22 @@ public function testGetFunctions() {
...
+    $filters = $this->twiggery->getFilters();
+    $this->assertInstanceOf(\Twig_SimpleFilter::class, $filters[0]);
+    $this->assertEquals('render_clean', $filters[0]->getName());
+  }
+
+  /**
+   * Run the render_clean filter.
+   */
+  public function testsCleanFilter() {
+    $markup = '<div><a><span><article><section>test</section></article></span></a>';
```
Comment (sherakama) [inline tests/src/Unit/Plugin/TwigPlugin/JumpstartUITwigTest.php:76]: Perhaps a few special characters in here would be good too.

**Comment by sherakama** [comment]
This looks good and super helpful. Do you have a testing plan to make sure we haven't undone something? Perhaps dig up some past tickets to use as a kickoff point for testing. 

It would be good to merge this just after a release so that it spends more time in developer/tester hands before getting a tag.

File: tests/src/Unit/Plugin/TwigPlugin/JumpstartUITwigTest.php (hunk 2 above)
Comment (pookmish) [reply tests/src/Unit/Plugin/TwigPlugin/JumpstartUITwigTest.php:76]: I believe using the `#markup` does thay for us. I tested a few characters out

**Comment by pookmish** [comment]
Yes I think we can definitely wait to merge this until after a release. It has the potential to really break stuff

File: src/Plugin/TwigPlugin/JumpstartUITwig.php (hunk 1 above)
Comment (pookmish) [reply src/Plugin/TwigPlugin/JumpstartUITwig.php:84]: Meh. I dont see much either way. Honestly I still haven't seen that tag. I've just trusted some documentation

**Comment by pookmish** [comment]
FTR: For D9 readiness, this PR is a requirement. `convert_encoding('UTF-8', 'HTML-ENTITIES')` will produce errors like: `Notice: iconv(): Wrong charset, conversion from 'HTML-ENTITIES' to 'UTF-8' is not allowed in twig_convert_encoding()`

File: tests/src/Unit/Plugin/TwigPlugin/JumpstartUITwigTest.php (hunk 2 above)
Comment (pookmish) [reply tests/src/Unit/Plugin/TwigPlugin/JumpstartUITwigTest.php:76]: Putting special characters in here are not modified in any way. by placing the resulting string into the array with `#markup` twig and drupal will do the decoding at that point. As long as this filter is not appended with `|raw` or something, there should be no issues.

**Comment by pookmish** [comment]
@sherakama any object to this since we are now lock into our next release. We will have almost exactly 1 month to test out this change.

File: composer.json [hunk 3]
```diff
@@ -13,9 +13,9 @@
     }
   ],
   "require": {
-    "drupal/core": "^8.8",
-    "drupal/components": "~1.0",
-    "drupal/ds": "~3.3",
-    "drupal/ui_patterns": "~1.0"
+    "drupal/core": "^8.8 || ^9.0",
+    "drupal/components": "^1.0 || ^2.0",
```
Comment (sherakama) [inline composer.json:17]: Should we force the 2.0 upgrade and punch a new version?

**Review by sherakama — APPROVED** [review]
The sooner we get this into others' hands the better. Thanks for this work.

File: composer.json (hunk 3 above)
Comment (pookmish) [reply composer.json:17]: The 2.0 is for the new version of twig that comes with drupal 9. So we can't force that upgrade at this time.

**Pull request merged**